src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/CustomerPreferences/CustomerPreferencesFormHandler.php

```diff
@@ -45,11 +45,11 @@
      */
     public function save(array $data)
     {
-        if (empty($this->dataProvider->setData($data))) {
+        $errors = parent::save($data);
+
+        if (empty($errors)) {
             $this->handleB2bUpdate($data['general']['enable_b2b_mode']);
         }
-
-        return parent::save($data);
```

Comment:
You maybe need to return `$errors` like the parent method, otherwise no need to affect the parent return into variable:

```php
if (empty(parent::save($data))) {
```
 (But I really think you need to return this variable :smiley: )